dask/dataframe/io/io.py

```diff
@@ -646,24 +650,25 @@ def to_records(df):
     return df.map_partitions(M.to_records)
 
 
-# TODO: type this -- causes lots of papercuts
 @insert_meta_param_description
```

Comment:
We'll need to add type annotations here as well, ref: https://github.com/dask/dask/runs/7159429564?check_suite_focus=true#step:4:67